iris_markdown_EH.Rmd

```diff
@@ -0,0 +1,123 @@
+---
+title: "Ethan's Iris Analysis in RMarkdown"
+author: "Ethan Hill"
+date: "2020-01-16"
+output: html_document
+---
+
+Hello! Here are the plots and the code for the plots I have generated with the Iris dataset. I have also gone ahead and included the workflow I used to get the final plots...
+
+
+It is always a good habit to set your working directory:
+
+```{r}
+setwd("C:/Users/hille/Documents/Rbootcamp_S2020/Iris-repo/")
+```
+
+
+Next, I've scrambled the iris dataset (gotta make it harder) and stored the scrambled data into the object called mix:
+
+```{r}
+oo <- sample(x=1:150, 150, replace=FALSE)
+mix <- iris[oo,]
+```
+
+
+Once I have the scrambled data stored in the object called "mix" I can store each of the columns as a seperate vector:
+
+```{r}
+species <- mix$Species
+petalL <- mix$Petal.Length
+petalW <- mix$Petal.Width
+sepalL <- mix$Sepal.Length
+sepalW <- mix$Sepal.Width
+```
+
+
+Now that I have each of the columns stored as seperate vectors, I can take the species vector and assign each species its own unique color based on logical operations:
+
+```{r}
+species2 <- species
+color2 <- rep("red", 150)
+species2 == "setosa"
+color2[species2 == "setosa"] <- "purple"
+color2
+color2[species2 == "versicolor"] <- "blue"
+color2
+color2[species2 == "virginica"] <- "orange"
+color2
+```
+
+Cool, Now we can start doing some real analyses. 
+I have have made a best fit line of regression for Sepel Width as a function of Petal Length and have displayed the output stats:
+
+```{r}
+sepalW.lm <- lm(sepalW ~ petalL)
+sepalW.lm
+```
+
+
+This is a scatter plot for "Petal Length vs Petal Width" with a line of best fit.
+```{r}
+plot(petalL, sepalW, col=color2, cex=1.5, pch=19, xlab="Petal Length", ylab="Sepal Width")
+abline(sepalW.lm, col="black")
+legend('top', legend = levels(species), col = c("purple", "blue", "orange"), cex = .8, pch = 19)
+```
+
+As we can see from the graph above, the "general" trend is that as petal length increases, the sepal width decreases.
+
+BUT WAIT THERE'S MORE!!!!
+
+
+We can also index the iris dataset by species and make a regression line for Sepel Width as a function of Petal Length FOR EACH SPECIES and we can ALSO calcuate error for the linear regression FOR EACH SPECIES:
+```{r]
+setosa <- mix[mix$Species == "setosa",]
+setosa
+setosaPLvsSW.lm <- lm(setosa$Sepal.Width ~ setosa$Petal.Length)
+setosaPLvsSW.lm
+b.setosa <- coef(setosaPLvsSW.lm)
+int.setosa <- b.setosa[1]
+slope.setosa <- b.setosa[2]
+yhat.setosa <- slope.setosa*setosa$Petal.Length+int.setosa
+yhat.setosa
+error.setosa.sepalW <- setosa$Sepal.Width-yhat.setosa
+error.setosa.sepalW
```

Comment:
This is very nice!!!  I love your idea to compute the regression separately by species.
But FYI if you just want to plot the line of best fit, there is already a function for that:
abline( setosaPLvsSW.lm)